autofetch/views.py

```diff
@@ -0,0 +1,137 @@
+from django.shortcuts import render
+from external.craigslist import fetch_craigslist_housing
+from external.googleapi.fetch import fetch_reverse_geocode
+from location.models import Location
+from location.models import Apartment
+from datetime import datetime
+from celery import shared_task
+from external.googleapi.g_utils import normalize_us_address
+import googlemaps
+
+
+CITY_LIST = ["brx", "brk", "fct", "lgi", "mnh", "jsy", "que", "stn", "wch"]
+
+
+def autofetch(request):
+
+    # The huge query: /autofetch/?city=brx,brk,fct,lgi,mnh,jsy,que,stn,wch&limit=1000000
+    if request.method == "GET":
+        # allow get param in url /autofetch/?city=brk,brx&limit=100
+
+        city_list, city, limit = set(), list(), 100
+
+        if "city" in request.GET:
+            city = request.GET.get("city").split(",")
+
+        if "limit" in request.GET:
+            limit = request.GET.get("limit")
+
+        for c in city:
+            if c in CITY_LIST:
+                city_list.add(c)
+
+        city_list = sorted(
+            city_list
+        )  # sorted for pass the test by having foreseeable order
+
+        bckgrndfetch.delay(list(city_list), int(limit))
+
+    return render(request, "account.html")
+
+
+@shared_task
+def bckgrndfetch(city_list, limit):
+
+    print(f"Start: {str(datetime.now())} \n")
+
+    for city_name in city_list:
+
+        print(f"Fetching {city_name}............")
+
+        try:
+            results = fetch_craigslist_housing(
+                limit=limit, site="newyork", category="apa", area=city_name
+            )
+
+        except AttributeError:
+            print(
+                f" Cannot write into city: {city_name}, because one of the url not exist\n"
+            )
+            continue
+
+        for r in results:
+
+            if len(Apartment.objects.filter(c_id=r["id"])) > 0:
+                continue
```

Comment:
maybe i'm reading this wrong, but won't this mean that if the retrieved apartment already exists in our db it will never get updated?